### PR TITLE
Error handling [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by rulles13**

### CODE REVIEW

File: app/components/DepartureRow.tsx [hunk 1]
```diff
@@ -7,34 +7,16 @@ function DepartureRow(props: any) {
     const {departure} = props    
 
     return(
-        <div className="departure-row">
+        departure
```
Comment (heleneruelle) [inline app/components/DepartureRow.tsx]: So you should either do some conditionnal rendering within a container div like 
```
return (
<div>
{
departure
? ....
: ....
}
</div>
)

```
or with 2 different return like 
```
if (departure) {
return (<div>...</div>)
} else {
return (<div>...</div>)
}
```

but no conditionnal check just after `render`

File: app/components/DepartureTimeHr.tsx [hunk 2]
```diff
@@ -6,9 +6,11 @@ function DepartureTimeHr(props: { departure: any }) {
     const departureTime = new Date(departureDateTime).toLocaleTimeString('fr-EU', { hour: '2-digit', minute: '2-digit' })
 
     return (
-        <div>
+        departureTime
+        ? (<div>
             {departureTime}    
-        </div>
+        </div>)
+        : (<p>??:??</p>)
```
Comment (heleneruelle) [inline app/components/DepartureTimeHr.tsx:13]: In this case the parent should have prevented the component from rendering 
Also departureTime will never be null, it'll be a date as you wanted or an `Invalid Date`

File: app/components/Headsign.tsx [hunk 3]
```diff
@@ -5,9 +5,12 @@ function Headsign(props: { departure: any }) {
     const {headsign} = departure
 
     return (
+        headsign
+        ? (
         <div>
             {headsign}    
-        </div>
+        </div>)
```
Comment (heleneruelle) [inline app/components/Headsign.tsx:10]: In this case the parent should have prevented the component from rendering

File: app/components/DepartureRow.tsx [hunk 4]
```diff
@@ -7,34 +7,16 @@ function DepartureRow(props: any) {
...
+        ? (<div className="departure-row">
```
Comment (heleneruelle) [inline app/components/DepartureRow.tsx]: You should check for departure.departureDateTime && departure.headsign (mandatory for your user) before rendering the children comps

File: app/pages/SmartDisplay.tsx [hunk 5]
```diff
@@ -19,13 +19,18 @@ function SmartDisplay() {
...
       <div className="main-container">
-        <Header stopDetails={stopDetails}/> 
-        <ul>
-          {departures.map((departure: any, index: number) => (
-            <DepartureRow key={index} departure={departure}/>
-            
-          ))}
-        </ul>
+        {
+          stopDetails
```
Comment (heleneruelle) [inline app/pages/SmartDisplay.tsx:23]: So this means if the stopDetails call failed, you are not displaying the departures (they are 2 different calls, one can fail but the other might have gone through, you want to display as many avail data as possible as long as they can make sense on their own)

**Review by heleneruelle — CHANGES_REQUESTED** [review]

File: app/components/DepartureTimeHr.tsx (hunk 2 above)
Comment (rulles13) [reply app/components/DepartureTimeHr.tsx:13]: I propose to use this to check the validity of the Date : 
`Date.parse(departureDateTime)`

File: app/components/Headsign.tsx (hunk 3 above)
Comment (rulles13) [reply app/components/Headsign.tsx:10]: i will remove the control.

File: app/pages/SmartDisplay.tsx (hunk 5 above)
Comment (rulles13) [reply app/pages/SmartDisplay.tsx:23]: In this case : I assume if no stop station no departure.

**Pull request merged**